pkg/storage/ocs.go

```diff
@@ -0,0 +1,415 @@
+}
+
+// WithManageNodes sets the storageCluster's managedNodes value.
+func (builder *StorageClusterBuilder) WithManageNodes(expectedManagedNodesValue bool) *StorageClusterBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof(
+		"Setting storageCluster %s in namespace %s with managedNodes value: %v",
+		builder.Definition.Name, builder.Definition.Namespace, expectedManagedNodesValue)
+
+	builder.Definition.Spec.ManageNodes = expectedManagedNodesValue
+
+	return builder
+}
+
+// WithManagedResources sets the storageCluster's managedResources value.
+func (builder *StorageClusterBuilder) WithManagedResources(
+	expectedManagedResources ocsoperatorv1.ManagedResourcesSpec) *StorageClusterBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof(
+		"Setting storageCluster %s in namespace %s with managedResources value: %v",
+		builder.Definition.Name, builder.Definition.Namespace, expectedManagedResources)
+
+	builder.Definition.Spec.ManagedResources = expectedManagedResources
+
+	return builder
+}
+
+// WithMonDataDirHostPath sets the storageCluster's monDataDirHostPath value.
+func (builder *StorageClusterBuilder) WithMonDataDirHostPath(
+	expectedMonDataDirHostPath string) *StorageClusterBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof(
+		"Setting storageCluster %s in namespace %s with monDataDirHostPath value: %v",
+		builder.Definition.Name, builder.Definition.Namespace, expectedMonDataDirHostPath)
+
+	if expectedMonDataDirHostPath == "" {
+		glog.V(100).Infof("the expectedMonDataDirHostPath can not be empty")
+
+		builder.errorMsg = "the expectedMonDataDirHostPath can not be empty"
+
+		return builder
+	}
+
+	builder.Definition.Spec.MonDataDirHostPath = expectedMonDataDirHostPath
+
+	return builder
+}
+
+// WithMultiCloudGateway sets the storageCluster's multiCloudGateway value.
+func (builder *StorageClusterBuilder) WithMultiCloudGateway(
+	expectedMultiCloudGateway ocsoperatorv1.MultiCloudGatewaySpec) *StorageClusterBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof(
+		"Setting storageCluster %s in namespace %s with multiCloudGateway value: %v",
+		builder.Definition.Name, builder.Definition.Namespace, expectedMultiCloudGateway)
+
+	builder.Definition.Spec.MultiCloudGateway = &expectedMultiCloudGateway
+
+	return builder
+}
+
+// WithStorageDeviceSet sets the storageCluster's storageDeviceSets value.
+func (builder *StorageClusterBuilder) WithStorageDeviceSet(
+	expectedStorageDeviceSet ocsoperatorv1.StorageDeviceSet) *StorageClusterBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof(
+		"Setting storageCluster %s in namespace %s with storageDeviceSets value: %v",
+		builder.Definition.Name, builder.Definition.Namespace, expectedStorageDeviceSet)
+
+	if builder.Definition.Spec.StorageDeviceSets == nil {
```

Comment:
No need for this condition. If it's nil append on line 385 will add new element.